From c981042e5c237bb8d3f5208b44c4995e2823dae3 Mon Sep 17 00:00:00 2001
From: Herrie <Github.com@herrie.org>
Date: Wed, 17 Jan 2024 09:09:31 +0100
Subject: [PATCH] WAM: WebPageBlink: keep same window type when creating
 additional windows

If an app has a specific "defaultWindowType" in appinfo.json, if should
be kept when opening an additional window.

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>
---
Upstream-Status: Inappropriate [LuneOS specific]

 src/platform/webengine/web_page_blink.cc | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/platform/webengine/web_page_blink.cc b/src/platform/webengine/web_page_blink.cc
index aa8e049..063e70f 100644
--- a/src/platform/webengine/web_page_blink.cc
+++ b/src/platform/webengine/web_page_blink.cc
@@ -1297,7 +1297,8 @@ WebView* WebPageBlink::CreateWindow(const std::string& newUrl, std::unique_ptr<W
 
   // Create a new webApp instance for this page
   WebAppManager *webAppMgr = WebAppManager::Instance();
-  webAppMgr->CreateWindowForAppPage(kWtCard, app_desc_, "{}", app_id_, newPage);
+  webAppMgr->CreateWindowForAppPage(webAppMgr->WindowTypeFromString(app_desc_->DefaultWindowType()), 
+                                    app_desc_, "{}", app_id_, newPage);
 
   return newPage->PageView();
 }
\ No newline at end of file
